src/server/browserContext.ts

```diff
@@ -71,7 +79,31 @@ export abstract class BrowserContext extends SdkObject {
   }
 
   async _initialize() {
-    await this.instrumentation.onContextCreated(this);
+    if (this.attribution.isInternal)
+      return;
+    // Create instrumentation per context.
+    this.instrumentation = createInstrumentation();
+
+    // Debugger will pause execution upon page.pause in headed mode.
+    const contextDebugger = new Debugger(this);
+    this.instrumentation.addListener(contextDebugger);
+
+    if (this._options._traceDir)
+      this.instrumentation.addListener(new Tracer(this, this._options._traceDir));
+
+
+    // When PWDEBUG=1, show inspector for each context.
+    if (debugMode() === 'inspector')
+      RecorderSupplement.showInspector(this);
+
+    // When paused, show inspector.
+    if (contextDebugger.isPaused())
+      RecorderSupplement.showInspector(this);
+    contextDebugger.on(Debugger.Events.PausedStateChanged, () => {
+      RecorderSupplement.showInspector(this);
+    });
+
+    await this.instrumentation.onContextCreated();
```

Comment:
Should we call this before various `showInspector`, so that all instrumentation listeners are ready? In anyone still listening to onContextCreated?

Reply:
RecorderSupplement.showInspector is now largely async to everything else. It'll query debugger state and subscribe to its events.